common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationActivity.java

```diff
@@ -60,6 +72,21 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
                         getIntent().getExtras().getString(SerializableSpanContext.SERIALIZABLE_SPAN_CONTEXT),
                         SerializableSpanContext.class
                 );
+                // Extract OtelContext from the carrier if it exists
+                final Map<String, String> carrier;
+                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.TIRAMISU) {
+                    carrier = getIntent().getExtras() != null ?
+                            getIntent().getExtras().getSerializable(OTEL_CONTEXT_CARRIER, HashMap.class) :
+                            new HashMap<>();
+                } else {
+                    @SuppressWarnings("unchecked")
+                    Map<String, String> temp = getIntent().getExtras() != null ?
+                            (Map<String, String>) getIntent().getExtras().getSerializable(OTEL_CONTEXT_CARRIER) :
+                            new HashMap<>();
+                    carrier = temp;
+                }
```

Comment:
Can you add some comments here describng why we need different logic based on OS version?

Reply:
Also it looks mostly same to me except the cast

Reply:
Added a comment. It is slightly safer, as the new Serializable interface performs type checking at the framework level (older one is deprecated now).